### PR TITLE
issue_3321_seperate_firebaseapp

### PR DESCRIPTION
**Pull request by argzdev**
Proposed fix for #3321 by dev-gloomyfox.

TLDR: FirebaseModelDownloader instance creation fails when using a separate FirebaseApp instance. 

The proposed fix will overload the `getInstance` method of both `FirebaseMlLogger.java` & `ModelFileManager.java`. This method will have a parameter of `FirebaseApp`, and will be used in [FirebaseModelDownloader.java#L58](https://github.com/firebase/firebase-android-sdk/blob/master/firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/FirebaseModelDownloader.java#L58) & [ModelFileDownloadService.java#L88](https://github.com/firebase/firebase-android-sdk/blob/master/firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/internal/ModelFileDownloadService.java#L88). 

More details is explained by the developer in the said issue.

### CODE REVIEW

File: firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/internal/ModelFileManager.java [hunk 1]
```diff
@@ -62,6 +63,12 @@ public static ModelFileManager getInstance() {
     return FirebaseApp.getInstance().get(ModelFileManager.class);
   }
 
+  @NonNull
+  public static ModelFileManager getInstance(FirebaseApp app) {
+    Preconditions.checkArgument(app != null, "Null is not a valid value of FirebaseApp.");
```
Comment (rlazo) [inline firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/internal/ModelFileManager.java:68]:
```suggestion
  public static ModelFileManager getInstance(@NonNull FirebaseApp app) {
```

File: firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/internal/ModelFileManager.java [hunk 2]
```diff
@@ -24,6 +24,7 @@
 import androidx.annotation.VisibleForTesting;
 import androidx.annotation.WorkerThread;
 import com.google.firebase.FirebaseApp;
+import com.google.firebase.components.Preconditions;
```
Comment (rlazo) [inline firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/internal/ModelFileManager.java:27]:
```suggestion
```

File: firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/internal/FirebaseMlLogger.java [hunk 3]
```diff
@@ -93,6 +94,12 @@ public static FirebaseMlLogger getInstance() {
     return FirebaseApp.getInstance().get(FirebaseMlLogger.class);
   }
 
+  @NonNull
+  public static FirebaseMlLogger getInstance(FirebaseApp app) {
+    Preconditions.checkArgument(app != null, "Null is not a valid value of FirebaseApp.");
```
Comment (rlazo) [inline firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/internal/FirebaseMlLogger.java:99]:
```suggestion
  public static FirebaseMlLogger getInstance(@NonNull FirebaseApp app) {
```

File: firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/internal/FirebaseMlLogger.java [hunk 4]
```diff
@@ -23,6 +23,7 @@
 import androidx.annotation.WorkerThread;
 import com.google.android.datatransport.TransportFactory;
 import com.google.firebase.FirebaseApp;
+import com.google.firebase.components.Preconditions;
```
Comment (rlazo) [inline firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/internal/FirebaseMlLogger.java:26]:
```suggestion
```

File: firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/internal/ModelFileManager.java [hunk 5]
```diff
@@ -64,8 +63,7 @@ public static ModelFileManager getInstance() {
   }
 
   @NonNull
-  public static ModelFileManager getInstance(FirebaseApp app) {
-    Preconditions.checkArgument(app != null, "Null is not a valid value of FirebaseApp.");
+  public static ModelFileManager getInstance(@NonNull FirebaseApp app) {
     return FirebaseApp.getInstance(app.getName()).get(ModelFileManager.class);
```
Comment (rlazo) [inline firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/internal/ModelFileManager.java:67]: `app.get(...)` should do the trick. See https://github.com/firebase/firebase-android-sdk/blob/ac3276ce7d5f5181b1c419731495eb88a98537af/firebase-segmentation/src/main/java/com/google/firebase/segmentation/FirebaseSegmentation.java#L72 for reference

Comment (argzdev) [reply firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/internal/ModelFileManager.java]: Thanks for the reference, so it should be like `return app.get(ModelFileManager.class);`?

**Review by rlazo — APPROVED** [review]
